ArcGISRuntimeSDKQt_CppSamples/Analysis/ViewshedLocation/ViewshedLocation.cpp

```diff
@@ -0,0 +1,414 @@
+}
+
+void ViewshedLocation::setVerticalAngle(double verticalAngle)
+{
+  if (m_locationViewshed)
+  {
+    if (m_locationViewshed->verticalAngle() == verticalAngle)
+      return;
+
+    m_veriticalAngle = verticalAngle;
+    m_locationViewshed->setVerticalAngle(verticalAngle);
+  }
+  else
+  {
+    if (m_veriticalAngle == verticalAngle)
+      return;
+
+    m_veriticalAngle = verticalAngle;
+  }
+
+  emit verticalAngleChanged();
+}
+
+double ViewshedLocation::heading() const
+{
+  return m_locationViewshed ? m_locationViewshed->heading() : m_heading;
+}
+
+void ViewshedLocation::setHeading(double heading)
+{
+  if (m_locationViewshed && m_locationViewshedGraphic)
+   {
+     if (m_locationViewshed->heading() == heading)
+       return;
+
+     m_heading = heading;
+     m_locationViewshed->setHeading(heading);
+
+     SimpleMarkerSceneSymbol* smss = static_cast<SimpleMarkerSceneSymbol*>(m_locationViewshedGraphic->symbol());
+     smss->setHeading(m_locationViewshed->heading()-180);
+   }
+   else
+   {
+     if (m_heading == heading)
+       return;
+
+     m_heading = heading;
+   }
+
+   emit headingChanged();
+}
+
+double ViewshedLocation::pitch() const
+{
+  return m_locationViewshed ? m_locationViewshed->pitch() : m_pitch;
+}
+
+void ViewshedLocation::setPitch(double pitch)
+{
+  if (m_locationViewshed)
+  {
+    if (m_locationViewshed->pitch() == pitch)
+      return;
+
+    m_pitch = pitch;
+    m_locationViewshed->setPitch(pitch);
+
+    SimpleMarkerSceneSymbol* smss = static_cast<SimpleMarkerSceneSymbol*>(m_locationViewshedGraphic->symbol());
+    smss->setPitch(qAbs(m_locationViewshed->pitch()-180));
+  }
+  else
+  {
+    if (m_pitch == pitch)
+      return;
+
+    m_pitch = pitch;
+  }
+
+  emit pitchChanged();
+}
+
+QColor ViewshedLocation::visibleColor() const
+{
+  return m_locationViewshed ? m_locationViewshed->visibleColor() : m_visibleColor;
+}
+
+void ViewshedLocation::setVisibleColor(const QColor& visibleColor)
+{
+  if (m_locationViewshed)
+  {
+    if (m_locationViewshed->visibleColor() == visibleColor)
+      return;
+
+    m_visibleColor = visibleColor;
+    m_locationViewshed->setVisibleColor(visibleColor);
+  }
+  else
+  {
+    if (m_visibleColor == visibleColor)
+      return;
+
+    m_visibleColor = visibleColor;
+  }
+
+  emit visibleColorChanged();
+}
+
+QColor ViewshedLocation::obstructedColor() const
+{
+  return m_locationViewshed ? m_locationViewshed->obstructedColor() : m_obstructedColor;
+}
+
+void ViewshedLocation::setObstructedColor(const QColor& obstructedColor)
+{
+  if (m_locationViewshed)
+  {
+    if (m_locationViewshed->obstructedColor() == obstructedColor)
+      return;
+
+    m_obstructedColor = obstructedColor;
+    m_locationViewshed->setObstructedColor(obstructedColor);
+  }
+  else
+  {
+    if (m_obstructedColor == obstructedColor)
+      return;
+
+    m_obstructedColor = obstructedColor;
+  }
+
+  emit obstructedColorChanged();
```

Comment:
The color methods should use the methods like they are static:
`Viewshed::visibleColor()`, `Viewshed::obstructedColor()`, etc...